pkg/prometheus/promcfg.go

```diff
@@ -216,12 +218,12 @@ func generateServiceMonitorConfig(version semver.Version, m *v1.ServiceMonitor,
 	switch version.Major {
 	case 1:
 		if version.Minor < 7 {
-			cfg = append(cfg, k8sSDAllNamespaces())
+			cfg = append(cfg, generateK8SSDConfig(nil, nil, nil))
```

Comment:
Am I right that if a user defines a custom API server config, and chooses to use Prometheus < 1.7 that user would not see any log messages, saying that the configuration is ignored? If so, could you add a log line here to inform the user?

Reply:
You're right. This configuration is ignored when using Prometheus < 1.7. I will add log messages for this.